### PR TITLE
Support for OSX and fix some bugs

### PR DESCRIPTION
**Pull request by xauthulei**
Support for OSX.
Fix bugs: 
 Split the PR# and Commit.
 Accurate matching the PR#. e.g: If the PR# is 1, it will search 10, 12 100, etc

### CODE REVIEW

File: prin [hunk 1]
```diff
@@ -53,7 +53,13 @@ PROG=${0##*/}
 # If NO ARGUMENTS should return *usage*, uncomment the following line:
 usage=${1:-yes}
 
-source $(dirname $(readlink -ne $BASH_SOURCE))/lib/common.sh
+if [[ $(uname) == "Darwin" ]]; then
```
Comment (spiffxp) [inline prin]: I find I can't run this with my >4.0 version of bash as installed by homebrew unless I invoke as `bash ./prin` or change the hashbang to `#!/usr/bin/env bash`

Comment (spiffxp) [reply prin]: There doesn't appear to be a check for whether `envsubst` exists

File: prin [hunk 2]
```diff
@@ -135,15 +141,25 @@ gitlib::current_branch >/dev/null || common::exit 1
...
                         --all --pretty=format:"%h"))
+elif [[ "${POSITIONAL_ARGV[0]}" =~ ^[a-zA-Z0-9]*$ ]]; then
+    COMMIT=${POSITIONAL_ARGV[0]}
+    INITIAL_COMMIT=$(git log --pretty=format:"%H - %an, %ar : %s" \
+				|grep "$COMMIT" \
+				|grep "Merge pull request" \
+				|cut -d'-' -f1)
+    CP_COMMIT=$(git log --pretty=format:"%H - %an, %ar : %s" \
+				|grep "$COMMIT" \
+				|grep "cherry-pick-of-"
```
Comment (spiffxp) [inline prin]:
```suggestion
				|grep "cherry-pick-of-"  \
```

File: prin (hunk 1 above)
Comment (xauthulei) [reply prin]: Thanks @spiffxp, That is real good suggestion. I will update it

File: prin (hunk 2 above)
Comment (xauthulei) [reply prin]: Updated,Thanks

File: prin (hunk 1 above)
Comment (spiffxp) [reply prin]: `envsubst` isn't present by default on darwin, can you check if it's present and error out if not?

**Comment by spiffxp** [comment]
/ok-to-test
/hold
Thanks for the update.  Missed one thing

File: prin (hunk 1 above)
Comment (xauthulei) [reply prin]: @spiffxp  Yea. added the check action here. please review this. Thanks for your great efforts.

**Comment by justaugustus** [comment]
@xauthulei -- the `pull-release-verify` is failing due to the verify-shellcheck target that was recently added (#726).
@pswica is working on cleaning up the bash scripts.

In the meantime, these failures can be safely ignored.

**Comment by xauthulei** [comment]
@justaugustus ,Got it. Thanks for this notice.

**Comment by xauthulei** [comment]
@justaugustus , because I am a newer here, what should I do for this PR currently. Many thanks.

File: prin [hunk 3]
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
```
[the remark this answers is not in the document]
Comment (xauthulei) [reply prin]: > Any reason for this change to `#!/usr/bin/env bash`?
This is from the suggestion of @spiffxp in previous review comments :
I find I can't run this with my >4.0 version of bash as installed by homebrew unless I invoke as bash ./prin or change the hashbang to #!/usr/bin/env bash

File: prin [hunk 4]
```diff
@@ -53,7 +53,14 @@ PROG=${0##*/}
 # If NO ARGUMENTS should return *usage*, uncomment the following line:
 usage=${1:-yes}
 
-source $(dirname $(readlink -ne $BASH_SOURCE))/lib/common.sh
```
Comment (pswica) [inline prin]: I think this move would cause issues if this is PR'd after I clean `./prin`. I'll make this the next one I clean against `shellcheck`

Comment (pswica) [reply prin]: https://github.com/kubernetes/release/pull/763

This is my PR to clean prin

**Comment by justaugustus** [comment]
@xauthulei -- hold off on rebasing; #766 needs to merge first as well.

File: prin (hunk 4 above)
Comment (xauthulei) [reply prin]: @pswica In fact, I don't remove it, By the way I just added the "if" condition before it. Also I merge yours and fixed some `shellcheck` error, please review that.

**Comment by xauthulei** [comment]
> @xauthulei -- hold off on rebasing; #766 needs to merge first as well.
I have merged the #766 and #763,  and fixed some `shellcheck` errors.  Currently, there is no `shellcheck` errors from 'prin' in "pull-release-verify" , Could you review that. thanks and what steps should I take now. Thanks

File: prin (hunk 4 above)
Comment (pswica) [reply prin]: Great. This looks good from a shellcheck perspective. Thanks a lot for your patience while I clean our scripts against shellcheck

Comment (xauthulei) [reply prin]: Thanks again for your great efforts.

**Comment by justaugustus** [comment]
Thanks for working on this, @xauthulei!
/hold cancel
/lgtm
/approve

**Pull request merged**